contributors/design-proposals/sig-cli/apply_v2.md

```diff
@@ -0,0 +1,201 @@
+# Apply v2
+
+## Background
+
+`kubectl apply` reads a file or set of files, and updates the cluster state based off the file contents.
+It does a couple things:
+
+1. Create / Update / (Delete) the resource based on the resources
+2. Only set / delete fields defined in the file either currently or in during last update from the file
+  - Retain updates to fields made by controllers that were never managed in the file
+3. Merge fields based on the field name, merge lists based on a field contained within the list elements
+
+The apply logic has been difficult to maintain from a contributor perspective, and difficult
+to understand from a user perspective.  Much of this pain has comes from the `apply` command out
+growing its implementation.  In order to continue to support apply and provide better user visibility
+into what it is doing, the command must be re-architected to break client dependencies on the server-side
```

Comment:
I thought we agreed that client-server was not the main issue.

Reply:
In order to make apply sufficiently maintainable and extensible to new API types, as well as to make its behavior more intuitive for users, the merge behavior, including how it is specified in the API schema, must be systematically redesigned and more thoroughly tested.